### PR TITLE
chore: validate permissions boundary policy name so as to fail faster if invalid

### PR DESCRIPTION
**Pull request by huanjani**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by paragbhingre — APPROVED** [review]

File: internal/pkg/aws/iam/iam.go [hunk 1]
```diff
@@ -98,6 +99,21 @@ func (c *IAM) CreateECSServiceLinkedRole() error {
 	return nil
 }
 
+// ListPermBoundPolicyNames returns a list of permissions boundary policy names.
+func (c *IAM) ListPermBoundPolicyNames() ([]*string, error) {
+	policies, err := c.client.ListPolicies(&iam.ListPoliciesInput{
```
Comment (iamhopaul123) [inline internal/pkg/aws/iam/iam.go]: We might need to handle pagination here: https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#ListPoliciesOutput

File: internal/pkg/aws/iam/iam.go [hunk 2]
```diff
@@ -98,6 +99,21 @@ func (c *IAM) CreateECSServiceLinkedRole() error {
...
+		PolicyUsageFilter: aws.String("PermissionsBoundary"),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("list IAM policies: %w", err)
+	}
+	var policyNames = make([]*string, len(policies.Policies))
```
Comment (iamhopaul123) [inline internal/pkg/aws/iam/iam.go]: Why not `[]string` 🤔

File: internal/pkg/aws/iam/iam.go (hunk 1 above)
Comment (huanjani) [reply internal/pkg/aws/iam/iam.go]: Yeah. I was hoping the `PolicyUsageFilter: aws.String("PermissionsBoundary")` filter would reduce the number to a reasonable amount, but I just tested it and you're right. Will add!

File: internal/pkg/aws/iam/iam.go (hunk 2 above)
Comment (huanjani) [reply internal/pkg/aws/iam/iam.go]: Changed.

File: internal/pkg/aws/iam/iam.go (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/aws/iam/iam.go]: yeah basically we need to do pagination for every list API we call as long as the response is paginated and we return the full list of it (instead of grabbing first xyz items)

File: internal/pkg/aws/iam/iam.go [hunk 3]
```diff
@@ -98,6 +99,32 @@ func (c *IAM) CreateECSServiceLinkedRole() error {
...
 }
 
+// ListPermBoundPolicyNames returns a list of permissions boundary policy names.
+func (c *IAM) ListPermBoundPolicyNames() ([]string, error) {
+	var policies []*iam.Policy
+	var marker *string
+	for {
+		output, err := c.client.ListPolicies(&iam.ListPoliciesInput{
+			Marker:            marker,
+			Scope:             aws.String("Local"),
```
Comment (iamhopaul123) [inline internal/pkg/aws/iam/iam.go:109]: How is this config related to "returns a list of permissions boundary policy names."?

File: internal/pkg/cli/app_init.go [hunk 4]
```diff
@@ -234,6 +242,19 @@ func (o *initAppOpts) validateAppName(name string) error {
...
+	IAMPolicies, err := o.iam.ListPermBoundPolicyNames()
+	if err != nil {
+		return fmt.Errorf("list permissions boundary policies: %w", err)
+	}
+	for _, policy := range IAMPolicies {
+		if policy == policyName {
+			return nil
+		}
+	}
+	return fmt.Errorf("IAM policy '%s' not found in this account", policyName)
```
Comment (iamhopaul123) [inline internal/pkg/cli/app_init.go]:
```suggestion
	return fmt.Errorf("IAM policy %q not found in this account", policyName)
```

File: internal/pkg/aws/iam/iam.go (hunk 3 above)
Comment (huanjani) [reply internal/pkg/aws/iam/iam.go:109]: changing the wording

**Review by iamhopaul123 — APPROVED** [review]
